@caesarxuchao 

```
panic: runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/runtime/panic.go:481 +0x3ff
/usr/local/go/src/testing/testing.go:467 +0x22a
/usr/local/go/src/runtime/panic.go:443 +0x521
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/garbagecollector/uid_cache.go:41 +0x121
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/garbagecollector/garbagecollector.go:414 +0xc05
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/controller/garbagecollector/garbagecollector_test.go:307 +0x24cd
/usr/local/go/src/testing/testing.go:473 +0xdd
/usr/local/go/src/testing/testing.go:582 +0xae3
```

Test failure seems related - can you please take a look?

--- a/pkg/controller/garbagecollector/garbagecollector_test.go
+++ b/pkg/controller/garbagecollector/garbagecollector_test.go
@@ -298,8 +298,9 @@ func TestProcessEvent(t *testing.T) {
 				uidToNode: make(map[types.UID]*node),
 			},
 			gc: &GarbageCollector{
-				dirtyQueue: workqueue.NewTimedWorkQueue(),
-				clock:      clock.RealClock{},
+				dirtyQueue:       workqueue.NewTimedWorkQueue(),
+				clock:            clock.RealClock{},
+				absentOwnerCache: NewUIDCache(2),
 			},
 		}
 		for i := 0; i < len(scenario.events); i++ {

